src/circuit/layouter.rs

```diff
@@ -67,58 +68,43 @@ pub trait RegionLayouter<C: Chip>: fmt::Debug {
 }
 
 /// A [`Layouter`] for a single-chip circuit.
-pub struct SingleChip<'a, C: Chip, CS: Assignment<C::Field> + 'a> {
+pub struct SingleChipLayouter<'a, F: FieldExt, CS: Assignment<F> + 'a> {
     cs: &'a mut CS,
-    config: C::Config,
-    loaded: Option<C::Loaded>,
     /// Stores the starting row for each region.
     regions: Vec<RegionStart>,
     /// Stores the first empty row for each column.
     columns: HashMap<Column<Any>, usize>,
-    _marker: PhantomData<C>,
+    marker: PhantomData<F>,
 }
 
-impl<'a, C: Chip, CS: Assignment<C::Field> + 'a> fmt::Debug for SingleChip<'a, C, CS> {
+impl<'a, F: FieldExt, CS: Assignment<F> + 'a> fmt::Debug for SingleChipLayouter<'a, F, CS> {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        f.debug_struct("SingleChip")
-            .field("config", &self.config)
+        f.debug_struct("SingleChipLayouter")
             .field("regions", &self.regions)
             .field("columns", &self.columns)
             .finish()
     }
 }
 
-impl<'a, C: Chip, CS: Assignment<C::Field>> SingleChip<'a, C, CS> {
+impl<'a, F: FieldExt, CS: Assignment<F>> SingleChipLayouter<'a, F, CS> {
     /// Creates a new single-chip layouter.
-    pub fn new(cs: &'a mut CS, config: C::Config) -> Result<Self, Error> {
-        let mut ret = SingleChip {
+    pub fn new(cs: &'a mut CS) -> Result<Self, Error> {
+        let ret = SingleChipLayouter {
             cs,
-            config,
-            loaded: None,
             regions: vec![],
             columns: HashMap::default(),
-            _marker: PhantomData,
+            marker: PhantomData,
```

Comment:
Undo this change (markers are never used, so to prevent lint errors):
```suggestion
            _marker: PhantomData,
```